### PR TITLE
Configure user and group in security context

### PR DESCRIPTION
**Pull request by frsann**
We set the user and group id in the security context to match the user and group defined explicitly in the docker images.

Note, the `console` uses [user id 1000](https://github.com/flyteorg/flyteconsole/blob/a64e1385852125bec9efe7f37c1d3930c7e2a434/Dockerfile#L30), while the rest of the images use user id `1001` and group id `1001` (see PRs below)

Related PRs:
* https://github.com/flyteorg/flytepropeller/pull/356
* https://github.com/flyteorg/flyteadmin/pull/280
* https://github.com/flyteorg/datacatalog/pull/54

Related issue:
* https://github.com/flyteorg/flyte/issues/1606

### CODE REVIEW

File: charts/flyte-core/templates/admin/deployment.yaml [hunk 1]
```diff
@@ -17,6 +17,10 @@ spec:
         {{- end }}
       labels: {{ include "flyteadmin.labels" . | nindent 8 }}
     spec:
+      securityContext:
+        fsGroup: 1001
+        runAsUser: 1001
+        fsGroupChangePolicy: "OnRootMismatch"
       initContainers:
         {{- if .Values.db.checks }}
         - name: check-db-ready
```
Comment (frsann) [inline charts/flyte-core/templates/admin/deployment.yaml:26]: I dont have a easy way of testing how this user definition works together with the initContainer defined below.

**Comment by pmahindrakar-oss** [comment]
@frsann i tested this change and admin pod doesn't come up.
```
kubectl logs -f flyteadmin-5cbc955498-6svq5 -n flyte generate-secrets
Using config file:  [/etc/flyte/config/cluster_resources.yaml /etc/flyte/config/db.yaml /etc/flyte/config/domain.yaml /etc/flyte/config/logger.yaml /etc/flyte/config/server.yaml /etc/flyte/config/storage.yaml /etc/flyte/config/task_resource_defaults.yaml]
Error: failed to create path [/etc/secrets/auth]. Error: mkdir /etc/secrets: permission denied
failed to create path [/etc/secrets/auth]. Error: mkdir /etc/secrets: permission denied
Usage:

```

Since admin init container generate secrets requires write access to /etc folder it fails.
Instead of docker file switching to use `USER flyte` , would it better to just create the user and the group in Dockerfile and have the orchestration yaml file contain the security context to use for each init container and the main container.

I have taken that approach in the admin change here 
https://github.com/flyteorg/flyteadmin/pull/278/

cc : @EngHabu @kumare3

**Comment by pmahindrakar-oss** [comment]
Instead of using /etc/secrets/auth for the flyteadmin generate-secrets container , I was able to use ~/secrets/auth which doesn't require root permissions and the init container was able to run successfully.
This worked for kustomize based generated sandbox but it failed to work on helm generated sandbox due to subsitution issues. Helm generated file when applied was interpreting ~/secrets/auth as //secrets/auth and fails . Couldnt debug much why this was happening though.


Another issue is the init-container for check-db-ready uses pg_isready which fails to resolve the host 
```
postgres:5432 - no attempt
waiting for database
postgres:5432 - no attempt
waiting for database
```
Whereas the same init container when run with root user works fine . I believe this could be due to /etc/hosts file permissions.

These were the issue i found for flyteadmin kustomize , helm . Wasn't able to run flyte-core since that work is in progress  AFAIK

File: charts/flyte-core/templates/admin/deployment.yaml (hunk 1 above)
Comment (EngHabu) [reply charts/flyte-core/templates/admin/deployment.yaml:26]: I think the best way to do this is to:

```bash
flytectl sandbox start
helm list -a -A
helm uninstall -n flyte <helm chart name>
helm install flyte --debug ./charts/flyte -f ./charts/flyte/values-sandbox.yaml --create-namespace --namespace=flyte
```

and see what happens :-)

**Comment by EngHabu** [comment]
I manually verified the changes in a separate deployment. I believe we can move ahead with the change...

Can you rerun `make kustomize && make helm` and commit the changes? @frsann

**Comment by EngHabu** [comment]
It seems it's still complaining about changes in kustomize and helm... you might want to make sure you have the same versions of these tools as the ones used in CI

File: charts/flyte-core/templates/datacatalog/deployment.yaml [hunk 2]
```diff
@@ -25,6 +25,8 @@ spec:
       {{- if .Values.db.checks }}
       - name: check-db-ready
         image: postgres:10.16-alpine
+        securityContext:
```
Comment (EngHabu) [inline charts/flyte-core/templates/datacatalog/deployment.yaml:28]: Perfect... yes that's what needed.. do you mind replicating that to kustomize?

**Review by EngHabu — APPROVED** [review]

**Pull request merged**